pkg/docker/docker.go

```diff
@@ -165,3 +166,55 @@ func (c *Client) TarImage(writer io.Writer, image pkgutil.Image) error {
 	}
 	return nil
 }
+
+// ImageListWithFilePath compiles container image names based on all Dockerfiles found, considering excludes
+// according to following search pattern: **/Dockerfile*
+// Return value contains a map with image names and file path
+// Examples for image names with imageName testImage
+// * Dockerfile: `imageName`
+// * sub1/Dockerfile: `imageName-sub1`
+// * sub2/Dockerfile_proxy: `imageName-sub2-proxy`
```

Comment:
2 cases that might be a bit strange with this approach:

1. `containerImageName: myApplication` and 2 Dockerfiles under "docker/Dockerfile" and "docker/Dockerfile.debug" would result in the images `myApplication-docker` and `myApplication-docker-debug`, right? Is this really the desired outcome?

2. Refactoring my directory structure would result in a different image name being. pushed. So I should update my deployment?

Reply:
sure, the solution comes with some assumptions but requires rather little configuration.
It is based on a real-life example of multiple teams creating several microservice-based products.

If there was a need to enhance based on user feedback, one could always consider enhancing the capabilities.

The way as suggested in #3452 using a `map[string]interface{}` parameter is not supported properly.
Cobra only provides simple and list parameters via command line. 
Also configuration resolution might not properly work.

Reply:
>  It is based on a real-life example of multiple teams creating several microservice-based products.

I guess that is my main point. This PR feels like custom made for one specific use case.